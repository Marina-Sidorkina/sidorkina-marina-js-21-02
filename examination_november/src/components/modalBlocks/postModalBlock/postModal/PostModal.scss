.post-modal {
  position: fixed;
  left: 0;
  top: 0;
  justify-content: center;
  width: 100%;
  min-width: 320px;
  height: 100%;
  padding: 30px;
  background-color: rgba(0, 0, 0, 0.7);
  z-index: 10;
  display: flex;

  &__content {
    position: relative;
    box-sizing: border-box;
    width: 100%;
    height: 100%;
    max-width: 700px;
    max-height: 700px;
    padding: 15px;
    margin-top: 20px;
    background-color: #f2f2f2;
    border-radius: 15px;
    overflow-y: auto;
  }

  &__button {
    position: fixed;
    right: 10px;
    top: 10px;
    width: 30px;
    height: 30px;
    cursor: pointer;
    background-color: transparent;
    border: none;

    &:hover {
      opacity: 0.7;
    }
  }

  &__paginator {
    width: 100%;
    margin-top: 10px;
    text-align: right;
  }
}
