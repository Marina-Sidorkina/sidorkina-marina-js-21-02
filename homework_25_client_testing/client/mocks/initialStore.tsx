const initialStore = {
  userModal: {
    isOpened: false,
    values: {
      picture: '',
      name: 'John Doe',
      gender: 'male',
      dateOfBirth: {
        ruDateAndTime: '05 October 2011 14:48 UTC',
      },
      registerDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC',
      },
      id: '12345',
      email: 'test',
      phone: '55555'
    }
  },
  login: {
    data: {
      error: false,
      isLoading: false,
      inputValue: '',
      authorizedUserId: '123345',
      authorizedUserName: 'Name',
      authorizedUserPicture: ''
    }
  },
  registration: {
    values: {
      name: '',
      gender: '',
      dateOfBirth: '',
      email: '',
      phone: '',
    },
    isLoading: false,
    error: false
  },
  languageSelector: {
    value: 'ru'
  },
  postModal: {
    isOpened: false,
    currentPostId: '12345'
  },
  userPosts: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      posts: [{
        id: '12345',
        image: '',
        text: 'text',
        publishDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC'
        },
        owner: {
          firstName: 'John',
          lastName: 'Doe',
          picture: '',
          title: 'mr',
          id: '12345'
        }
      }]
    }
  },
  postsList: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      posts: [{
        id: '12345',
        image: '',
        text: 'text',
        publishDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC'
        },
        owner: {
          firstName: 'John',
          lastName: 'Doe',
          picture: '',
          title: 'mr',
          id: '12345'
        }
      }]
    }
  },
  usersList: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      users: [{
        firstName: 'John',
        lastName: 'Doe',
        title: 'mr',
        id: '12345'
      }]
    }
  },
  userInfo: {
    data: {
      isLoading: false,
      user: {
        firstName: 'John',
        lastName: 'Doe',
        picture: '',
        title: 'mr',
        gender: 'male',
        dateOfBirth: {
          ruDateAndTime: '05 October 2011 14:48 UTC',
        },
        registerDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC',
        },
        id: '12345',
        email: '',
        phone: '55555'
      }
    }
  },
  postModalPost: {
    isLoading: false,
    error: false,
    owner: {
      firstName: 'John',
      lastName: 'Doe',
      picture: '',
      title: 'mr',
      id: '12345'
    },
    post: {
      id: '12345',
      image: '',
      text: 'text',
      publishDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC'
      }
    }
  },
  postModalComments: {
    page: 1,
    error: false,
    isLoading: false,
    comments: [{
      id: '1234',
      message: 'message',
      publishDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC'
      }
    }],
    owners: [{
      firstName: 'John',
      lastName: 'Doe',
      picture: '',
      title: 'mr',
      id: '12345'
    }],
    limit: 6,
    totalComments: 0
  }
};

export default initialStore;
