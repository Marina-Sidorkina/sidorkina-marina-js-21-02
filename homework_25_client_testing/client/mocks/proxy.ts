import MockAdapter from 'axios-mock-adapter';
import { proxy } from '../src/api/proxy';

export const mock = new MockAdapter(proxy);
