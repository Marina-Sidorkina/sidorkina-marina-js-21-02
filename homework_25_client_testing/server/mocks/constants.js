module.exports = {
  errorText: 'Request failed with status code 520',
  successStatus: 200,
  errorStatus: 520
};
