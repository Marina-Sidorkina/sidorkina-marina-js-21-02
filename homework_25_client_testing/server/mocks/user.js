module.exports = {
  userListMockData: { data: [], page: 0, limit: 5, total: 100 },
  userMockData: {
    id: '60d0fe4f5311236168a109ce',
    title: 'mr',
    firstName: 'Rudi',
    lastName: 'Droste',
    picture: 'https://randomuser.me/api/portraits/med/men/83.jpg',
    gender: 'male',
    email: 'rudi.droste@example.com',
    dateOfBirth: '2021-06-21T21:02:08.029Z',
    phone: '0421-7342787',
    location: {
      street: '1196, MÃ¼hlenweg',
      city: 'GrÃ¼nhain-Beierfeld',
      state: 'Sachsen-Anhalt',
      country: 'Germany',
      timezone: '+3:30'
    },
    registerDate: '2021-06-21T21:02:08.029Z',
    updatedDate: '2021-06-21T21:02:08.029Z'
  },
  userProcessedMockData: {
    id: '60d0fe4f5311236168a109ce',
    firstName: 'Rudi',
    lastName: 'Droste',
    title: 'mr',
    gender: 'male',
    email: 'rudi.droste@example.com',
    phone: '0421-7342787',
    dateOfBirth: {
      enDate: '22 of June 2021',
      ruDate: '22 Июня 2021 года',
      enDateAndTime: '22 of June 00:02',
      ruDateAndTime: '22 Июня 00:02'
    },
    registerDate: {
      enDate: '22 of June 2021',
      ruDate: '22 Июня 2021 года',
      enDateAndTime: '22 of June 00:02',
      ruDateAndTime: '22 Июня 00:02'
    },
    picture: 'https://randomuser.me/api/portraits/med/men/83.jpg'
  },
  userPostsListMockData: {
    data: [{id: '12345', image: 'image', owner: {}, text: 'text'}],
    page: 0, limit: 1, total: 100
  },
  userUpdatedMockData: { firstName: 'Rudi', lastname: 'Droste', gender: 'male',
    email: 'rudi.droste@example.com', dateOfBirth: '2021-06-21T21:02:08.029Z',
    phone: '0421-7342787'
  }
};
