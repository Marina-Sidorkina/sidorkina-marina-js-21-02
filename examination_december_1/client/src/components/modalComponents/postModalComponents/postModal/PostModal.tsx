import React, { useContext } from 'react';
import { CloseOutlined } from '@ant-design/icons';
import { useDispatch } from 'react-redux';
import styles from './PostModal.module.scss';
import Paginator from '../../../commonComponents/paginator/Paginator';
import PostModalPost from '../postModalPost/PostModalPost';
import PostModalComments from '../postModalComments/PostModalComments';
import { closePostModalAction } from '../../../../redux/actions/postModal';
import { updatePostModalCommentsAction, updatePostModalCommentsPageAction }
  from '../../../../redux/actions/postModalComments';
import { ThemeContext } from '../../../../contexts/ThemeContext';
import { useTypedSelector } from '../../../../redux/hooks/useTypedSelector';
import { COMMENTS_PER_PAGE, DEFAULT_PAGES_AMOUNT } from '../../../../constants/components';

const PostModal = () => {
  const themeContext = useContext(ThemeContext);
  const stateValues = useTypedSelector((state) => state);
  const dispatch = useDispatch();

  return (
    <div className={styles.modal}>
      <button
        type="button"
        className={styles.button}
        onClick={() => {
          dispatch(updatePostModalCommentsAction([]));
          dispatch(closePostModalAction());
        }}
      >
        <CloseOutlined style={{ color: '#ffffff', fontSize: '25px' }} />
      </button>
      <div className={themeContext.darkTheme
        ? `${styles.content} ${styles.content_dark}`
        : styles.content}
      >
        <PostModalPost />
        <PostModalComments />
        <div className={styles.paginator}>
          <Paginator
            current={stateValues.postModalComments.page}
            total={stateValues.postModalComments.totalComments || DEFAULT_PAGES_AMOUNT}
            perPage={COMMENTS_PER_PAGE}
            onPageChange={(newPage: number) => {
              dispatch(updatePostModalCommentsPageAction(newPage));
            }}
            modal
          />
        </div>
      </div>
    </div>
  );
};

export default PostModal;
