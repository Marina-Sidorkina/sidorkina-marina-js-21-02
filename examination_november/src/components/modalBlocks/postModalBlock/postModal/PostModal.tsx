import React from 'react';
import './PostModal.scss';
import { CloseOutlined } from '@ant-design/icons';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Paginator from '../../../commonBlocks/paginator/Paginator';
import PostModalPost from '../postModalPost/PostModalPost';
import PostModalComments from '../postModalComments/PostModalComments';
import { closePostModalAction } from '../../../../redux/actions/postModal';

interface IPostModalProps {
  closeModal: Function;
}

const PostModal = (props: IPostModalProps) => {
  const { closeModal } = props;

  return (
    <div className="post-modal">
      <button
        type="button"
        className="post-modal__button"
        onClick={() => closeModal()}
      >
        <CloseOutlined style={{ color: '#ffffff', fontSize: '25px' }} />
      </button>
      <div className="post-modal__content">
        <PostModalPost />
        <PostModalComments />
        <div className="post-modal__paginator">
          <Paginator
            current={1}
            total={50}
            perPage={4}
            onPageChange={(page: number) => console.log(page)}
            modal
          />
        </div>
      </div>
    </div>
  );
};

export default connect(
  () => ({}),
  (dispatch) => ({
    closeModal: bindActionCreators(closePostModalAction, dispatch)
  })
)(PostModal);
