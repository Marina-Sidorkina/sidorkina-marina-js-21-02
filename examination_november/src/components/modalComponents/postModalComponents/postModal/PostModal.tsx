import React, { useContext } from 'react';
import './PostModal.scss';
import { CloseOutlined } from '@ant-design/icons';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Paginator from '../../../commonComponents/paginator/Paginator';
import PostModalPost from '../postModalPost/PostModalPost';
import PostModalComments from '../postModalComments/PostModalComments';
import { closePostModalAction } from '../../../../redux/actions/postModal';
import {
  updatePostModalCommentsAction,
  updatePostModalCommentsPageAction
} from '../../../../redux/actions/postModalComments';
import { ThemeContext } from '../../../../contexts/ThemeContext';

interface IPostModalProps {
  closeModal: Function;
  updatePostModalComments: Function;
  page: number;
  total: number;
  updatePostModalCommentsPage: Function;
}

const PostModal = (props: IPostModalProps) => {
  const {
    closeModal, updatePostModalComments, page, total, updatePostModalCommentsPage
  } = props;

  const themeContext = useContext(ThemeContext);

  return (
    <div className="post-modal">
      <button
        type="button"
        className="post-modal__button"
        onClick={() => {
          updatePostModalComments([]);
          closeModal();
        }}
      >
        <CloseOutlined style={{ color: '#ffffff', fontSize: '25px' }} />
      </button>
      <div className={`${themeContext.darkTheme
        ? 'post-modal__content post-modal__content_dark'
        : 'post-modal__content'}`}
      >
        <PostModalPost />
        <PostModalComments />
        <div className="post-modal__paginator">
          <Paginator
            current={page}
            total={total || 1}
            perPage={6}
            onPageChange={(newPage: number) => updatePostModalCommentsPage(newPage)}
            modal
          />
        </div>
      </div>
    </div>
  );
};

export default connect(
  (state: any) => ({
    page: state.postModalComments.page,
    total: state.postModalComments.totalComments,
  }),
  (dispatch) => ({
    closeModal: bindActionCreators(closePostModalAction, dispatch),
    updatePostModalComments: bindActionCreators(updatePostModalCommentsAction, dispatch),
    updatePostModalCommentsPage: bindActionCreators(updatePostModalCommentsPageAction, dispatch)
  })
)(PostModal);
