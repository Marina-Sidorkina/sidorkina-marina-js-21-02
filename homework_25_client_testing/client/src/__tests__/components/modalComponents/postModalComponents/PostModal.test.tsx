import '../../../../../mocks/matchMediaMock';
import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import PostModal from '../../../../components/modalComponents/postModalComponents/postModal/PostModal';
import mockStore from '../../../../../mocks/mockStore';
import initialStore from '../../../../../mocks/initialStore';

configure({ adapter: new Adapter() });

describe('PostModal test', () => {
  let store: any;
  let wrapper: any;

  beforeEach(() => {
    store = mockStore(initialStore);
    wrapper = mount(
      <Provider store={store}>
        <HashRouter>
          <PostModal />
        </HashRouter>
      </Provider>
    );
  });

  test('should render PostModal', () => {
    expect(wrapper.length).toEqual(1);
  });

  test('should render PostModal', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
