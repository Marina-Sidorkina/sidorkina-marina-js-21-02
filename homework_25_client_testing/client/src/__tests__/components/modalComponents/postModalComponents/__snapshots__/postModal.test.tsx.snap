// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostModal test should render PostModal 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <PostModal>
        <div
          className="modal"
        >
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            <ForwardRef(CloseOutlined)
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": "25px",
                }
              }
            >
              <AntdIcon
                icon={
                  Object {
                    "icon": Object {
                      "attrs": Object {
                        "focusable": "false",
                        "viewBox": "64 64 896 896",
                      },
                      "children": Array [
                        Object {
                          "attrs": Object {
                            "d": "M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z",
                          },
                          "tag": "path",
                        },
                      ],
                      "tag": "svg",
                    },
                    "name": "close",
                    "theme": "outlined",
                  }
                }
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": "25px",
                  }
                }
              >
                <span
                  aria-label="close"
                  className="anticon anticon-close"
                  role="img"
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": "25px",
                    }
                  }
                >
                  <IconReact
                    icon={
                      Object {
                        "icon": Object {
                          "attrs": Object {
                            "focusable": "false",
                            "viewBox": "64 64 896 896",
                          },
                          "children": Array [
                            Object {
                              "attrs": Object {
                                "d": "M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z",
                              },
                              "tag": "path",
                            },
                          ],
                          "tag": "svg",
                        },
                        "name": "close",
                        "theme": "outlined",
                      }
                    }
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      key="svg-close"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                        key="svg-close-svg-0"
                      />
                    </svg>
                  </IconReact>
                </span>
              </AntdIcon>
            </ForwardRef(CloseOutlined)>
          </button>
          <div
            className="content"
          >
            <PostModalPost>
              <div
                className="modalPost"
              >
                <div
                  className="userInfo"
                >
                  <div
                    className="user"
                  >
                    <img
                      alt="Аватар пользователя"
                      className="userImg"
                      src=""
                    />
                    <ComponentWithHelper
                      firstName="John"
                      lastName="Doe"
                    >
                      <div
                        className="helper"
                        onMouseOut={[Function]}
                        onMouseOver={[Function]}
                      >
                        <PostModalPostName
                          firstName="John"
                          lastName="Doe"
                        >
                          <div
                            className="userName"
                          >
                            John Doe
                          </div>
                        </PostModalPostName>
                      </div>
                    </ComponentWithHelper>
                  </div>
                  <div
                    className="postDate"
                  >
                    05 October 2011 14:48 UTC
                  </div>
                </div>
                <img
                  alt="Пост пользователя"
                  className="img"
                  src=""
                />
                <p
                  className="text"
                >
                  text
                </p>
              </div>
            </PostModalPost>
            <PostModalComments>
              <ul
                className="comments"
              >
                <PostModalCommentsItem
                  date={
                    Object {
                      "ruDateAndTime": "05 October 2011 14:48 UTC",
                    }
                  }
                  firstName="John"
                  id="12345"
                  img=""
                  key="1234"
                  lastName="Doe"
                  text="message"
                >
                  <li
                    className="comment"
                  >
                    <img
                      alt="Аватар пользователя"
                      className="img"
                      src=""
                    />
                    <div
                      className="info"
                    >
                      <div
                        className="user"
                      >
                        <ComponentWithHelper
                          firstName="John"
                          lastName="Doe"
                        >
                          <div
                            className="helper"
                            onMouseOut={[Function]}
                            onMouseOver={[Function]}
                          >
                            <PostModalCommentsItemName
                              firstName="John"
                              lastName="Doe"
                            >
                              <div
                                className="name"
                              >
                                John Doe
                              </div>
                            </PostModalCommentsItemName>
                          </div>
                        </ComponentWithHelper>
                        <div
                          className="date"
                        >
                          05 October 2011 14:48 UTC
                        </div>
                      </div>
                      <p
                        className="text"
                      >
                        message
                      </p>
                    </div>
                  </li>
                </PostModalCommentsItem>
              </ul>
            </PostModalComments>
            <div
              className="paginator"
            >
              <Paginator
                current={1}
                modal={true}
                onPageChange={[Function]}
                perPage={6}
                total={1}
              >
                <Pagination
                  className="paginator paginator_modal
        "
                  current={1}
                  onChange={[Function]}
                  pageSize={6}
                  showQuickJumper={true}
                  simple={true}
                  total={1}
                >
                  <LocaleReceiver
                    componentName="Pagination"
                    defaultLocale={
                      Object {
                        "items_per_page": "/ page",
                        "jump_to": "Go to",
                        "jump_to_confirm": "confirm",
                        "next_3": "Next 3 Pages",
                        "next_5": "Next 5 Pages",
                        "next_page": "Next Page",
                        "page": "Page",
                        "page_size": "Page Size",
                        "prev_3": "Previous 3 Pages",
                        "prev_5": "Previous 5 Pages",
                        "prev_page": "Previous Page",
                      }
                    }
                  >
                    <Pagination
                      className="paginator paginator_modal
        "
                      current={1}
                      defaultCurrent={1}
                      defaultPageSize={10}
                      hideOnSinglePage={false}
                      itemRender={[Function]}
                      jumpNextIcon={
                        <a
                          className="ant-pagination-item-link"
                        >
                          <div
                            className="ant-pagination-item-container"
                          >
                            <ForwardRef(DoubleRightOutlined)
                              className="ant-pagination-item-link-icon"
                            />
                            <span
                              className="ant-pagination-item-ellipsis"
                            >
                              •••
                            </span>
                          </div>
                        </a>
                      }
                      jumpPrevIcon={
                        <a
                          className="ant-pagination-item-link"
                        >
                          <div
                            className="ant-pagination-item-container"
                          >
                            <ForwardRef(DoubleLeftOutlined)
                              className="ant-pagination-item-link-icon"
                            />
                            <span
                              className="ant-pagination-item-ellipsis"
                            >
                              •••
                            </span>
                          </div>
                        </a>
                      }
                      locale={
                        Object {
                          "items_per_page": "/ page",
                          "jump_to": "Go to",
                          "jump_to_confirm": "confirm",
                          "next_3": "Next 3 Pages",
                          "next_5": "Next 5 Pages",
                          "next_page": "Next Page",
                          "page": "Page",
                          "page_size": "Page Size",
                          "prev_3": "Previous 3 Pages",
                          "prev_5": "Previous 5 Pages",
                          "prev_page": "Previous Page",
                        }
                      }
                      nextIcon={
                        <button
                          className="ant-pagination-item-link"
                          tabIndex={-1}
                          type="button"
                        >
                          <ForwardRef(RightOutlined) />
                        </button>
                      }
                      onChange={[Function]}
                      onShowSizeChange={[Function]}
                      pageSize={6}
                      prefixCls="ant-pagination"
                      prevIcon={
                        <button
                          className="ant-pagination-item-link"
                          tabIndex={-1}
                          type="button"
                        >
                          <ForwardRef(LeftOutlined) />
                        </button>
                      }
                      selectComponentClass={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "OptGroup": [Function],
                          "Option": [Function],
                          "SECRET_COMBOBOX_MODE_DO_NOT_USE": "SECRET_COMBOBOX_MODE_DO_NOT_USE",
                          "render": [Function],
                        }
                      }
                      selectPrefixCls="ant-select"
                      showLessItems={false}
                      showPrevNextJumpers={true}
                      showQuickJumper={true}
                      showTitle={true}
                      simple={true}
                      style={Object {}}
                      total={1}
                      totalBoundaryShowSizeChanger={50}
                    >
                      <ul
                        className="ant-pagination ant-pagination-simple paginator paginator_modal
        "
                        style={Object {}}
                      >
                        <li
                          aria-disabled={true}
                          className="ant-pagination-prev ant-pagination-disabled"
                          onClick={[Function]}
                          onKeyPress={[Function]}
                          tabIndex={null}
                          title="Previous Page"
                        >
                          <button
                            className="ant-pagination-item-link"
                            disabled={true}
                            tabIndex={-1}
                            type="button"
                          >
                            <ForwardRef(LeftOutlined)>
                              <AntdIcon
                                icon={
                                  Object {
                                    "icon": Object {
                                      "attrs": Object {
                                        "focusable": "false",
                                        "viewBox": "64 64 896 896",
                                      },
                                      "children": Array [
                                        Object {
                                          "attrs": Object {
                                            "d": "M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z",
                                          },
                                          "tag": "path",
                                        },
                                      ],
                                      "tag": "svg",
                                    },
                                    "name": "left",
                                    "theme": "outlined",
                                  }
                                }
                              >
                                <span
                                  aria-label="left"
                                  className="anticon anticon-left"
                                  role="img"
                                >
                                  <IconReact
                                    icon={
                                      Object {
                                        "icon": Object {
                                          "attrs": Object {
                                            "focusable": "false",
                                            "viewBox": "64 64 896 896",
                                          },
                                          "children": Array [
                                            Object {
                                              "attrs": Object {
                                                "d": "M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z",
                                              },
                                              "tag": "path",
                                            },
                                          ],
                                          "tag": "svg",
                                        },
                                        "name": "left",
                                        "theme": "outlined",
                                      }
                                    }
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="left"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      key="svg-left"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                                        key="svg-left-svg-0"
                                      />
                                    </svg>
                                  </IconReact>
                                </span>
                              </AntdIcon>
                            </ForwardRef(LeftOutlined)>
                          </button>
                        </li>
                        <li
                          className="ant-pagination-simple-pager"
                          title="1/1"
                        >
                          <input
                            onBlur={[Function]}
                            onChange={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            size="3"
                            type="text"
                            value={1}
                          />
                          <span
                            className="ant-pagination-slash"
                          >
                            /
                          </span>
                          1
                        </li>
                        <li
                          aria-disabled={true}
                          className="ant-pagination-next ant-pagination-disabled"
                          onClick={[Function]}
                          onKeyPress={[Function]}
                          tabIndex={null}
                          title="Next Page"
                        >
                          <button
                            className="ant-pagination-item-link"
                            disabled={true}
                            tabIndex={-1}
                            type="button"
                          >
                            <ForwardRef(RightOutlined)>
                              <AntdIcon
                                icon={
                                  Object {
                                    "icon": Object {
                                      "attrs": Object {
                                        "focusable": "false",
                                        "viewBox": "64 64 896 896",
                                      },
                                      "children": Array [
                                        Object {
                                          "attrs": Object {
                                            "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                                          },
                                          "tag": "path",
                                        },
                                      ],
                                      "tag": "svg",
                                    },
                                    "name": "right",
                                    "theme": "outlined",
                                  }
                                }
                              >
                                <span
                                  aria-label="right"
                                  className="anticon anticon-right"
                                  role="img"
                                >
                                  <IconReact
                                    icon={
                                      Object {
                                        "icon": Object {
                                          "attrs": Object {
                                            "focusable": "false",
                                            "viewBox": "64 64 896 896",
                                          },
                                          "children": Array [
                                            Object {
                                              "attrs": Object {
                                                "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                                              },
                                              "tag": "path",
                                            },
                                          ],
                                          "tag": "svg",
                                        },
                                        "name": "right",
                                        "theme": "outlined",
                                      }
                                    }
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="right"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      key="svg-right"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                                        key="svg-right-svg-0"
                                      />
                                    </svg>
                                  </IconReact>
                                </span>
                              </AntdIcon>
                            </ForwardRef(RightOutlined)>
                          </button>
                        </li>
                      </ul>
                    </Pagination>
                  </LocaleReceiver>
                </Pagination>
              </Paginator>
            </div>
          </div>
        </div>
      </PostModal>
    </Router>
  </HashRouter>
</Provider>
`;
